dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/debug/UiRumDebugListener.kt

```diff
@@ -109,8 +112,17 @@ internal class UiRumDebugListener(
     // region RumDebugListener
 
     override fun onReceiveActiveRumViews(viewNames: List<String>) {
-        rumViewsContainer?.post {
-            showRumViewsInfo(viewNames)
+        synchronized(viewsSnapshot) {
+            if (viewsSnapshot.isEmpty() ||
```

Comment:
as an extra optimization for the first condition you could add : `viewSnapshot.isEmpty()&&viewNames.isNotEmpty()`

Reply:
it can be the case when first time callback is called we have no active RUM views, then `viewNames` will be empty.